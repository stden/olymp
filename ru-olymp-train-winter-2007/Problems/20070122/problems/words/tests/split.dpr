{$APPTYPE CONSOLE}
Program Split;
uses SysUtils;

const TaskName = 'split';

var i, n, test: Integer;
 s, fname: String;
 fin, fout, ftest: Text;

BEGIN
 reset (fin, TaskName + '.in');
 reset (fout, TaskName + '.out');
 test := 0;
 SetLength (fname, 2);
 while NOT seekeof (fin) do begin
  inc (test);
  fname[1] := chr ((test mod 100) div 10 + 48);
  fname[2] := chr (test mod 10 + 48);
  rewrite (ftest, fname);
  readln (fin, s);
  writeln (ftest, s);
  close (ftest);
  rewrite (ftest, fname + '.a');
  readln (fout, n);
  writeln (ftest, n);
  for i := 1 to n do begin
   readln (fout, s);
   writeln (ftest, s);
  end;
  close (ftest);
 end;
END.
