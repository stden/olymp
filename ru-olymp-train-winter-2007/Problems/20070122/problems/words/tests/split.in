1 2 1
1 3 2
1 4 3
1 5 4
1 6 5
1 7 6
1 8 7
1 9 8
1 10 9
1 11 10
1 12 11
1 13 12
1 14 13
1 15 14
1 16 15
1 17 16
1 18 17
1 19 18
1 20 19
2 2 20
2 3 21
2 4 22
2 5 23
2 6 24
2 7 25
2 8 26
2 9 27
2 10 28
3 2 29
3 3 30
3 4 31
3 5 32
3 6 33
4 2 34
4 3 35
4 4 36
4 5 37
5 2 38
5 3 39
5 4 40
6 2 41
6 3 42
7 2 43
8 2 44
9 2 45
10 2 46
