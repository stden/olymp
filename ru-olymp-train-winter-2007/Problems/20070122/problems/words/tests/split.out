1
A
1
A
1
A
1
A
2
A
D
2
A
D
2
A
D
3
A
D
G
3
A
D
G
3
A
D
G
4
A
D
G
J
4
A
D
G
J
4
A
D
G
J
5
A
D
G
J
M
5
A
D
G
J
M
5
A
D
G
J
M
6
A
D
G
J
M
P
6
A
D
G
J
M
P
6
A
D
G
J
M
P
1
AA
1
AA
1
AA
3
AA
DB
BD
4
AA
DA
AD
DD
7
AA
DB
GC
CD
FE
BF
EG
8
AA
EA
DC
HC
CE
GE
BG
FG
9
AA
DA
GA
AD
DD
GD
AG
DG
GG
12
AA
DA
GB
JC
CD
FD
IE
BF
EG
HG
AH
HJ
1
AAA
1
AAA
4
AAA
DCB
CAC
BCD
6
AAA
DBA
BDA
EAC
BAD
CDD
16
AAA
DBA
CDA
EAB
AEB
CBC
BDC
EEC
DAD
ABD
FDD
BAE
DEE
FBF
EDF
BEF
1
AAAA
3
AAAA
CBAB
BCAC
8
AAAA
CABA
ADCB
CDDB
ACAC
CCBC
ABCD
CBDD
20
AAAA
DBAA
BDAA
EACA
CBCA
DDAB
CDCB
AECB
BEEB
ECAC
ACDC
EEEC
CEBD
BCCD
EADD
DCED
ACBE
DDCE
BADE
ECDE
2
AAAAA
BABAB
3
AAAAA
CBABA
BCACA
21
AAAAA
CABAA
CBABA
ABDBA
DDACA
BACCA
CCBAB
DCDAB
ADDCB
BCCDB
CACAC
AADAC
ABCBC
DAACC
BDCCC
CCDDC
BCBAD
DBBBD
BAACD
CDBCD
DCADD
2
AAAAAA
BABABA
9
AAAAAA
CBABAA
BCACAA
CBBAAB
BCBBAB
AABCAB
BCCAAC
AACBAC
CBCCAC
2
AAAAAAA
BABABAA
4
AAAAAAAA
BABABAAA
BAAAABAB
AABABBAB
6
AAAAAAAAA
BABABAAAA
BAAAABABA
AABAABAAB
ABAABBBAB
BBBABBABB
12
AAAAAAAAAA
BABABAAAAA
BAAAABABAA
AABAABAABA
ABAABBBABA
BBBABBABBA
ABBBABAAAB
BBAAABBAAB
BAABBBBBAB
BBAABBAABB
AAABABBABB
ABABBBABBB
